riscv/devices.cc

```diff
@@ -12,6 +12,22 @@ void bus_t::add_device(reg_t addr, abstract_device_t* dev)
   devices[addr] = dev;
 }
 
+void bus_t::set_interrupt_controller(abstract_interrupt_controller_t* intdev)
+{
+  intctrl = intdev;
+}
+
+abstract_interrupt_controller_t* bus_t::get_interrupt_controller(void)
+{
+  return intctrl;
+}
+
+void bus_t::set_interrupt_level(uint32_t interrupt_id, int level)
+{
+  if (intctrl)
```

Comment:
Is it a bug to call `set_interrupt_level` if `set_interrupt_controller` has not been called?  If so, should this be `assert(intctrl);` instead?